aws-cpp-sdk-transfer/source/transfer/UploadFileRequest.cpp

```diff
@@ -779,9 +751,17 @@ bool UploadFileRequest::ProcessAvailableBuffers()
 
     std::shared_ptr<UploadBuffer> thisBuffer;
 
-    while (!DoneWithRequests() && GetReadyBuffer(thisBuffer))
+    while (!DoneWithRequests() && !IsCancelled())
     {
-        ProcessBuffer(thisBuffer);
+        if (GetReadyBuffer(thisBuffer))
+        {
+            ProcessBuffer(thisBuffer);
+        }
+        else
+        {
+            // TODO: sleep/wait instead?

```

Comment:
this is waiting for an upload so it could be minutes, right ? in that case, this sort of code isn't great because it will take up close to 100% of a CPU. Sleep is risky because how long do you sleep for ?
So best would be a wait-post here. However, I am ok with leaving that for iteration 2 (i.e., if we have time).
For iteration (1), likely best to change to a small sleep (10 ms or something like that).